x/programs/runtime/runtime_test.go

```diff
@@ -57,3 +58,24 @@ func TestRuntimeCallProgramComplexReturn(t *testing.T) {
 	require.NoError(err)
 	require.Equal(ComplexReturn{Program: program.Address, MaxUnits: 1000}, into[ComplexReturn](result))
 }
+
+func TestContextInjection(t *testing.T) {
+	require := require.New(t)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	program := newTestProgram(ctx, "context_injection")
+
+	result, err := program.CallWithTimestamp(1, "get_timestamp")
+	require.NoError(err)
+	require.Equal(uint64(1), into[uint64](result))
+
+	result, err = program.CallWithHeight(1, "get_height")
+	require.NoError(err)
+	require.Equal(uint64(1), into[uint64](result))
+
+	result, err = program.CallWithActor(codec.CreateAddress(0, ids.GenerateTestID()), "get_actor")
+	require.NoError(err)
+	require.NotEqual(ids.Empty, into[ids.ID](result))
```

Comment:
These should be three different tests. You can use `t.Run` like I did in this PR to separate these into three different tests.

The issue is that if `CallWithTimeStamp` fails, you should still run `CallWithHeight` and `CallWithActor`. 

When anyone is debugging this, it's a lot more telling where the issue is if three different assertions fail rather than adding a patch only to get the first assertion to pass. Do you see what I mean? Unit tests are supposed to help with debugging as much as they're meant to prove that something works as expected.

Reply:
Thanks, yeah you are right that isolating unit tests makes them much easier to debug. I have used `t.Run` but it feels like the test is repeating. David's PR on default struct does not solve that because the function would panic if the default value was already set.
I have probably found a happy solution, passed a function pointer in one of the fields of the test property and executed the changing logic here.

Reply:
😅 
>like I did in this PR

I forgot to provide the link